domain/learningandworkprogress/src/main/kotlin/uk/gov/justice/digital/hmpps/domain/learningandworkprogress/conversation/service/ConversationService.kt

```diff
@@ -41,37 +42,40 @@ class ConversationService(
    * The order of the returned conversations is not guaranteed and it is up to the calling code to sort them
    * as necessary.
    */
-  fun getPrisonerConversations(prisonNumber: String): List<Conversation> {
+  fun getPrisonerConversations(prisonNumber: String, pageNumber: Int, pageSize: Int): PagedResult<Conversation> {
     log.info { "Retrieving Conversations for prisoner [$prisonNumber]" }
-    return persistenceAdapter.getConversations(prisonNumber)
+    return persistenceAdapter.getPagedConversations(prisonNumber, pageNumber, pageSize)
   }
 
   /**
-   * Returns a [Conversation] identified by its `conversationReference`.
-   * Throws [ConversationNotFoundException] if the [Conversation] cannot be found.
+   * Returns a [Conversation] identified by its `conversationReference` and validated against the associated
+   * `prisonNumber`.
+   * Throws [PrisonerConversationNotFoundException] if the [Conversation] cannot be found.
    */
-  fun getConversation(conversationReference: UUID): Conversation {
+  fun getConversation(conversationReference: UUID, prisonNumber: String): Conversation {
     log.info { "Retrieving Conversation with reference [$conversationReference]" }
-    return persistenceAdapter.getConversation(conversationReference)
-      ?: throw ConversationNotFoundException(conversationReference).also {
-        log.info { "Conversation with reference [$conversationReference] not found" }
-      }
+    return persistenceAdapter.getConversation(conversationReference)?.takeIf {
+      it.prisonNumber == prisonNumber
+    } ?: throw PrisonerConversationNotFoundException(conversationReference, prisonNumber).also {
```

Comment:
This `PrisonerConversationNotFoundException` is new and distinguishes between a conversation not existing and a prisoner conversation pair not existing.